### PR TITLE
pm: fix oopses when topology/firmware fails to load.

### PR DESCRIPTION
**Pull request by lgirdwood**
Topology loading failure or firmware boot failure can lead to crashes on some HW. Fix.

### CODE REVIEW

File: sound/soc/sof/pm.c [hunk 1]
```diff
@@ -143,6 +143,12 @@ static int sof_restore_pipelines(struct snd_sof_dev *sdev)
 		struct sof_ipc_reply reply;
 		struct sof_ipc_dai_config *config = dai->dai_config;
 
+		if (!config) {
+			dev_err(sdev->dev, "error: no config for DAI %s\n",
+				dai->name);
+			continue;
```
Comment (plbossart) [inline sound/soc/sof/pm.c:149]: ok

File: sound/soc/sof/pm.c [hunk 2]
```diff
@@ -240,6 +240,10 @@ static int sof_resume(struct device *dev, bool runtime_resume)
 	struct snd_sof_dev *sdev = dev_get_drvdata(&priv->pdev_pcm->dev);
 	int ret = 0;
 
+	/* check that sdev is valid */
+	if (!sdev)
+		return 0;
+
```
Comment (plbossart) [inline sound/soc/sof/pm.c]: no, this isn't needed. Ranjani and I worked on a better way to fix this with the removal of the sof-audio device.

**Comment by ranj063** [comment]
@lgirdwood with the changes proposed to remove sof-audio I think the second commit would probably not be needed. But the first commit looks good.

**Comment by lgirdwood** [comment]
@ranj063 @plbossart  I was expecting 2/2 to be superfluous since you are aligning the PM over to the PCI/ACPI device. I will keep 2/2 locally until your PM alignment patch lands. Updated.

**Pull request merged**